canonical-web-and-design/maas-ui
# refactor(ui): reorganise images components to follow fetching image flow

## Done

- Reorganised Images components so that there is a cleaner separation between synced images and changing the source of those images

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/images and check that a header displays where the images are synced from
- Click "Change source" and check that you can fetch images from either maas.io or a custom source

## Fixes

Fixes #2859 

## Screenshot
![2021-07-02_14-19](https://user-images.githubusercontent.com/25733845/124220016-832bae80-db40-11eb-827d-09171b3c8e14.png)
